docs/guides/upgrading-v3.md

```diff
@@ -196,6 +196,31 @@ That allowed us to make the KMS configuration consistent with all other AWS reso
 
 `alexaSkill` events now require an `appId` ([learn more](../deprecations.md#support-for-alexaskill-event-without-appid-is-to-be-removed)). That change was required to implement a more stable deployment, as well as to deploy more restricted IAM permissions.
 
+### Lambda Hashing Algorithm
+
+By default, Lambda version hashes will now be generated using improved algorithm (fixed determinism issues). As it is a breaking change that requires more manual effort during migration, it is still possible (but not recommended) to keep using old algorithm by using the following configuration:
+
+```
+provider:
+  lambdaHashingVersion: 20200924
+```
+
+However, we highly encourage an upgrade to the new algorithm. Safe upgrade requires change in configuration or code of all your functions. You can do it safely following the guide below:
```

Comment:
"Safe upgrade" makes me think (as a user) that something may not be safe (loosing stuff? destroying resources accidentally?)

I'd rephrase with something like:

```suggestion
However, we highly encourage an upgrade to the new algorithm. To upgrade, you must redeploy your service with code or configuration changes in all functions. You can do it by following the guide below:
```